File: java/src/mouseracer/Theme.java
package mouseracer;

import javax.swing.*;
import java.awt.*;

public class Theme {
    public static final Color BG_TOP = new Color(15, 15, 35);
    public static final Color BG_BOTTOM = new Color(30, 30, 60);
    public static final Color ACCENT = new Color(0, 200, 255);
    public static final Color ACCENT_DIM = new Color(0, 120, 180);
    public static final Color SURFACE_BORDER = new Color(60, 60, 100);
    public static final Color TEXT_PRIMARY = new Color(230, 235, 255);
    public static final Color TEXT_SECONDARY = new Color(140, 150, 180);
    public static final Color BTN_TOP = new Color(0, 180, 230);
    public static final Color BTN_BOTTOM = new Color(0, 100, 180);
    public static final Color BTN_DISABLED_TOP = new Color(50, 50, 70);
    public static final Color BTN_DISABLED_BOTTOM = new Color(35, 35, 55);
    public static final Color DANGER_TOP = new Color(180, 50, 50);
    public static final Color DANGER_BOTTOM = new Color(120, 30, 30);
    public static final Color SUCCESS = new Color(0, 220, 120);
    public static final Color GAME_OVER = new Color(255, 80, 80);
    public static final Color RESULTS_BG = new Color(20, 20, 45);

    public static final Color GOLD = new Color(255, 215, 0);
    public static final Color SILVER = new Color(192, 192, 192);
    public static final Color BRONZE = new Color(205, 127, 50);

    public static Color brighter(Color c, int amount) {
        return new Color(
            Math.min(255, c.getRed() + amount),
            Math.min(255, c.getGreen() + amount),
            Math.min(255, c.getBlue() + amount)
        );
    }

    public static Color withAlpha(Color c, float alpha) {
        int a = Math.max(0, Math.min(255, (int) (alpha * 255)));
        return new Color(c.getRed(), c.getGreen(), c.getBlue(), a);
    }

    public static JLabel styledLabel(String text, Color color, int size, int style) {
        JLabel label = new JLabel(text);
        label.setForeground(color);
        label.setFont(new Font("SansSerif", style, size));
        return label;
    }

    public static JButton styledButton(String text, Color bg) {
        JButton btn = new JButton(text);
        btn.setBackground(bg);
        btn.setForeground(Color.WHITE);
        btn.setFocusPainted(false);
        return btn;
    }
}

File: rust/src/ui/theme.rs
use eframe::egui::{self, Color32, Visuals};

pub const BG_TOP: Color32 = Color32::from_rgb(15, 15, 35);
pub const BG_BOTTOM: Color32 = Color32::from_rgb(30, 30, 60);
pub const ACCENT: Color32 = Color32::from_rgb(0, 200, 255);
pub const SURFACE_BORDER: Color32 = Color32::from_rgb(60, 60, 100);
pub const TEXT_PRIMARY: Color32 = Color32::from_rgb(230, 235, 255);
pub const TEXT_SECONDARY: Color32 = Color32::from_rgb(140, 150, 180);
pub const BTN_TOP: Color32 = Color32::from_rgb(0, 180, 230);
pub const BTN_BOTTOM: Color32 = Color32::from_rgb(0, 100, 180);
pub const BTN_DISABLED_TOP: Color32 = Color32::from_rgb(50, 50, 70);
pub const BTN_DISABLED_BOTTOM: Color32 = Color32::from_rgb(35, 35, 55);
pub const DANGER_TOP: Color32 = Color32::from_rgb(180, 50, 50);
pub const DANGER_BOTTOM: Color32 = Color32::from_rgb(120, 30, 30);
pub const SUCCESS: Color32 = Color32::from_rgb(0, 220, 120);
pub const RESULTS_BG: Color32 = Color32::from_rgb(20, 20, 45);
pub const GAME_OVER_COLOR: Color32 = Color32::from_rgb(255, 80, 80);

pub fn brighter(color: Color32, amount: u8) -> Color32 {
    Color32::from_rgb(
        color.r().saturating_add(amount),
        color.g().saturating_add(amount),
        color.b().saturating_add(amount),
    )
}

pub fn lerp_color(a: Color32, b: Color32, t: f32) -> Color32 {
    let t = t.clamp(0.0, 1.0);
    Color32::from_rgb(
        (a.r() as f32 + (b.r() as f32 - a.r() as f32) * t) as u8,
        (a.g() as f32 + (b.g() as f32 - a.g() as f32) * t) as u8,
        (a.b() as f32 + (b.b() as f32 - a.b() as f32) * t) as u8,
    )
}

pub fn configure_visuals(ctx: &egui::Context) {
    let mut visuals = Visuals::dark();
    visuals.panel_fill = BG_TOP;
    visuals.window_fill = Color32::from_rgb(20, 20, 45);
    visuals.override_text_color = Some(TEXT_PRIMARY);
    ctx.set_visuals(visuals);
}
